### PR TITLE
Observer parametric velocity deviation

### PR DESCRIPTION
**Pull request by juloberno**
- switch back to non frenet-based state deviation
- makes actually more sense since sensor noise is not influence by the current lane the vehicle is but instead mainly by the direction the object is moving to

### CODE REVIEW

File: bark/models/observer/observer_model_parametric.cpp [hunk 1]
```diff
@@ -49,36 +49,36 @@ ObservedWorld ObserverModelParametric::Observe(
...
-  // Get Current Frenet State of Agent
-  const Point2d pos = agent->GetCurrentPosition();
-  const auto& lane_corridor = agent->GetRoadCorridor()->GetCurrentLaneCorridor(pos);
-  if(!lane_corridor) {
-    return;
-  };
-  FrenetState current_frenet_state(agent->GetCurrentState(), lane_corridor->GetCenterLine());
-
+void ObserverModelParametric::AddStateDeviationFrenet(const AgentPtr& agent, const DistributionPtr& multi_dim_distribution,
+                         const ObservedWorldPtr& previous_observed_world, const double& delta_time) const {
```
Comment (patrickhart) [inline bark/models/observer/observer_model_parametric.cpp:68]: If we do not use delta_time, do we need this here?

File: bark/models/observer/observer_model_parametric.cpp [hunk 2]
```diff
@@ -49,36 +49,36 @@ ObservedWorld ObserverModelParametric::Observe(
...
-  current_frenet_state.lon += frenet_deviation[0];
-  current_frenet_state.lat += frenet_deviation[1];
-  
-  // Convert back and set dynamic agent state 
-  const auto deviated_state = FrenetStateToDynamicState(current_frenet_state, lane_corridor->GetCenterLine());
-  agent->SetCurrentState(deviated_state);
+  // Get Current Frenet State of Agent
+  auto state = agent->GetCurrentState();
+  const auto theta = state[StateDefinition::THETA_POSITION];
+  state[StateDefinition::X_POSITION] += state_deviation[0]*cos(theta) + state_deviation[1]*sin(theta);
```
Comment (patrickhart) [inline bark/models/observer/observer_model_parametric.cpp:76]: why do we need a transformation here? we can have a distribution that defines the deviations on the Cartesian coordinates

Comment (patrickhart) [reply bark/models/observer/observer_model_parametric.cpp:76]: Unless theta is used from the sampled values. so would need to update theta before and then calculate the x, y position

Comment (juloberno) [reply bark/models/observer/observer_model_parametric.cpp:76]: Deviations in Cartesian coordiantes would change the results if we rotate the whole scene. This formulation is independent of road placement in world coordinates.

Comment (patrickhart) [reply bark/models/observer/observer_model_parametric.cpp:76]: but theta would still be sampled, why not use the sampled one?

**Review by patrickhart — APPROVED** [review]

**Pull request merged**